annotations.md

```diff
@@ -28,7 +28,16 @@ This specification defines the following annotation keys, intended for but not l
 * **org.opencontainers.image.revision** Source control revision identifier for the packaged software.
 * **org.opencontainers.image.vendor** Name of the distributing entity, organization or individual.
 * **org.opencontainers.image.licenses** License(s) under which contained software is distributed as a [SPDX License Expression][spdx-license-expression].
-* **org.opencontainers.image.ref.name** Name of the reference for a target (string). SHOULD only be considered valid when on descriptors on `index.json` within [image layout](image-layout.md).
+* **org.opencontainers.image.ref.name** Name of the reference for a target (string).
+  * SHOULD only be considered valid when on descriptors on `index.json` within [image layout](image-layout.md).
+  * Character set of the value SHOULD conform to alphanum of `A-Za-z0-9` and separator set of `-._:@/+`
+  * An EBNF + regular expression like:
```

Comment:
In [XML 1.0's EBNF][1], this would be:

```
ref ::= component ("/" component)*
component ::= alphanum (separator alphanum)*
alphanum ::= [a-zA-Z0-9]+
separator ::= [-._:@+] | "--"
```

which is close to the current PR.  I think the XML EBNF spec is harder to read than the RFC 5234 or ISO/IEC 14977:1996 (although it's certainly shorter), and I'd rather lean on something stand-alone than a part of the XML spec, but even with those caveats I'd rather use the XML EBNF than rolling our own unspecified EBNF-ish.

[1]: https://www.w3.org/TR/REC-xml/#sec-notation